glfw/src/Window.zig

```diff
@@ -3221,3 +3221,104 @@ test "setScrollCallback" {
+    std.debug.print("note: failed to create window: {}\n", .{err});
+    return;
+}
+
+test "hint-attribute default value parity" {
+    try glfw.init(.{});
+    defer glfw.terminate();
+    const window_a = @as(Error!Window, handle: {
+        // manually create window, to avoid call to `Window.Hints.set` in `Window.create`, which
+        // would overwrite the default hint values set by GLFW itself.
+        const handle = c.glfwCreateWindow(640, 480, "Hello, Zig!", null, null);
+        getError() catch |err| break :handle err;
+        break :handle Window.from(handle.?) catch |err| break :handle err;
+    }) catch |err| return failedToCreateWindow(err);
+    defer window_a.destroy();
+
+    const window_b = Window.create(640, 480, "Hello, Zig!", null, null, .{}) catch |err| return failedToCreateWindow(err);
+    defer window_b.destroy();
+
+    inline for (comptime std.enums.values(Window.Hint)) |hint_tag| {
+        if (@hasField(Window.Attrib, @tagName(hint_tag))) {
+            const attrib_tag = @field(Window.Attrib, @tagName(hint_tag));
+            switch (attrib_tag) {
+                .resizable,
+                .visible,
+                .decorated,
+                .auto_iconify,
+                .floating,
+                .maximized,
+                .transparent_framebuffer,
+                .focus_on_show,
+                .client_api,
+                .context_creation_api,
+                .context_version_major,
+                .context_version_minor,
+                .context_robustness,
+                .context_release_behavior,
+                .opengl_forward_compat,
+                .opengl_debug_context,
+                .opengl_profile,
+                => {
+                    const expected = window_a.getAttrib(attrib_tag) catch |err| {
+                        std.debug.print("Failed to get attribute '{}' value from window_a with error '{}'.\n", .{ attrib_tag, err });
+                        return;
+                    };
+                    const actual = window_b.getAttrib(attrib_tag) catch |err| {
+                        std.debug.print("Failed to get attribute '{}' value from window_b with error '{}'.\n", .{ attrib_tag, err });
+                        return;
+                    };
+
+                    testing.expectEqual(expected, actual) catch |err| {
+                        std.debug.print("On attribute '{}'.\n", .{hint_tag});
+                        return err;
+                    };
+                },
+
+                // This attribute is based on a check for which window is currently in focus,
+                // and the default value, as of writing this comment, is 'true', which means
+                // that first window_a takes focus, and then window_b takes focus, meaning
+                // that we can't actually test for the default value.
+                .focused => continue,
+
+                // TODO: consider this one differently, as we've chosen a different default value
```

Comment:
What do you mean by this? "as we've chosen a different default value"?

Reply:
In the discussion [here](https://github.com/hexops/mach/pull/71#discussion_r748416222) in my window hints rework pr, me and @silversquirl came to the accord that, seeing as GLFW doesn't specify a default value for `context_no_error` in their docs in [this list](https://www.glfw.org/docs/latest/window_guide.html#window_hints_values), it would be most sane to default it to false (seeing as, in zig, we almost always want to handle errors with error codes, and avoid UB).

Though, thinking about it now, and looking at the source code, it actually is defaulted to false (albeit not directly, it's done through a call to `memset`), so there's no actual difference in behavior here. Though, maybe this calls for a pr upstream to add its default value to the aforementioned list in their docs? Then I could amend this "patch" in this pr.

Reply:
Ahh okay, I see. I was aware of the fact that it was implicitly set to false in GLFW through `memset` (I double checked the defaults when I reviewed, which was why I was surprised to see this TODO)

In that case, I am happy with this as-is and think we can just remove the TODO mention here.